GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -1459,6 +1463,28 @@ private void ResetSelectedItemsWithConfirmation(bool resetToParent)
 
         private static bool RenamedIndexItem(FileStatusItem item) => item.Item.IsRenamed && item.Item.Staged == StagedStatus.Index;
 
+        List<string> IRunScript.IFileListSource.GetFiles()
+        {
+            IReadOnlyList<FileStatusItem> selectedItems = DiffFiles.SelectedItems.ToList();
+
+            if (!selectedItems.Any())
+            {
+                return null;
```

Comment:
❗  It's a very bad practice to return `null` from methods that declare a a collection as a return type.

Reply:
`null` has a special meaning here, indicating that the script runner, down the line, should fail (pop up an error dialog) if it's expecting files (`{SelectedFiles}` is used, and is not discarded by an `{if:` block). I changed the signature to `IEnumerable<string>?`, to indicate `null` is a valid value, and also added an empty check downstream, for safety.